runescape-client-injector-plugin/src/main/java/net/runelite/injector/raw/ForcePlaySound.java

```diff
@@ -0,0 +1,187 @@
+import net.runelite.asm.attributes.code.instructions.GetStatic;
+import net.runelite.asm.attributes.code.instructions.IMul;
+import net.runelite.asm.attributes.code.instructions.IStore;
+import net.runelite.asm.attributes.code.instructions.InvokeStatic;
+import net.runelite.asm.attributes.code.instructions.LDC;
+import net.runelite.asm.signature.Signature;
+import net.runelite.deob.DeobAnnotations;
+import net.runelite.injector.Inject;
+import net.runelite.injector.InjectionException;
+import static net.runelite.injector.InjectHookMethod.HOOKS;
+
+public class ForcePlaySound
+{
+	private final Inject inject;
+
+	public ForcePlaySound(Inject inject)
+	{
+		this.inject = inject;
+	}
+
+	public void inject() throws InjectionException
+	{
+		injectForcePlaySound();
+	}
+
+	private void injectForcePlaySound() throws InjectionException
+	{
+		boolean injected = false;
+
+		Field soundEffectVolume = findStaticField("soundEffectVolume");
+
+		if (soundEffectVolume == null)
+		{
+			throw new InjectionException("Mapped field \"soundEffectVolume\" could not be found.");
+		}
+
+		net.runelite.asm.pool.Field poolSoundEffectVolume = soundEffectVolume.getPoolField();
+
+		for (ClassFile c : inject.getVanilla().getClasses())
+		{
+			for (Method m : c.getMethods())
+			{
+				if (m.getCode() == null)
+				{
+					continue;
+				}
+
+				Instructions instructions = m.getCode().getInstructions();
+
+				// Let's find "getstatic <some class>.soundEffectVolume":
+				// LDC -946689099
+				// GETSTATIC client.pf : I
+				// IMUL
+				// ISTORE 3
+
+				ListIterator<Instruction> iterator = instructions.getInstructions().listIterator();
+				while (iterator.hasNext())
+				{
+					Instruction i = iterator.next();
+
+					if (!(i instanceof GetStatic))
+					{
+						continue;
+					}
+
+					if (!iterator.hasNext())
+					{
+						break;
+					}
+
+					Instruction i2 = iterator.next();
+
+					if (!(i2 instanceof IMul) && !(i2 instanceof LDC))
+					{
+						iterator.previous();
+						continue;
+					}
+
+					if (!iterator.hasNext())
+					{
+						break;
+					}
+
+					Instruction i3 = iterator.next();
+
+					if (!(i3 instanceof IStore))
+					{
+						iterator.previous();
+						continue;
+					}
+
+					GetStatic gs = (GetStatic) i;
+
+					if (!gs.getField().equals(poolSoundEffectVolume))
+					{
+						continue;
+					}
+
+					InvokeStatic invoke = new InvokeStatic(instructions,
+						new net.runelite.asm.pool.Method(
+							new net.runelite.asm.pool.Class(HOOKS),
+							"getVolumeForQueuedSoundEffects",
+							new Signature("()I")
+						)
+					);
+
+					// Remove:
+					// LDC -946689099
+					// GETSTATIC client.pf : I
+					// IMUL
+
+					iterator.previous();
+					iterator.previous();
+					iterator.remove();
+					iterator.previous();
+					iterator.remove();
+					iterator.previous();
+					iterator.remove();
+
+					// Add invoke before "istore"
+
+					iterator.add(invoke);
+
+					injected = true;
```

Comment:
I assume this is all for the call to method2317? Is that inlined? Maybe we can replace it instead.